packages/components/src/components/combobox/combobox.component.ts

```diff
@@ -0,0 +1,967 @@
+    this.popup.active = false;
+
+    this.emit('syn-after-hide');
+  }
+
+  /**
+   * Shows the listbox. If it is not possible to open the listbox, because there are no
+   * appropriate filtered options, a syn-error is emitted and the listbox stays closed.
+   */
+  async show() {
+    if (this.open || this.disabled) {
+      this.open = false;
+      return undefined;
+    }
+
+    this.open = true;
+    return Promise.race([waitForEvent(this, 'syn-after-show'), waitForEvent(this, 'syn-error')]);
+  }
+
+  /** Hides the listbox. */
+  async hide() {
+    if (!this.open || this.disabled) {
+      this.open = false;
+      return undefined;
+    }
+
+    this.open = false;
+    return waitForEvent(this, 'syn-after-hide');
+  }
+
+  /**
+   * Checks for validity but does not show a validation message.
+   * Returns `true` when valid and `false` when invalid.
+   */
+  checkValidity() {
+    return this.valueInput.checkValidity();
+  }
+
+  /** Gets the associated form, if one exists. */
+  getForm(): HTMLFormElement | null {
+    return this.formControlController.getForm();
+  }
+
+  /** Checks for validity and shows the browser's validation message if the control is invalid. */
+  reportValidity() {
+    return this.valueInput.reportValidity();
+  }
+
+  /** Sets a custom validation message. Pass an empty string to restore validity. */
+  setCustomValidity(message: string) {
+    this.valueInput.setCustomValidity(message);
+    this.formControlController.updateValidity();
+  }
+
+  /** Sets focus on the control. */
+  focus(options?: FocusOptions) {
+    this.displayInput.focus(options);
+  }
+
+  /** Removes focus from the control. */
+  blur() {
+    this.displayInput.blur();
+  }
+
+  private createComboboxOptionsFromQuery(queryString: string) {
+    const optgroups: SynOptGroup[] = [];
+
+    this.filteredOptions = this.getSlottedOptions()
+      .filter(option => this.filter(option, queryString) || queryString === '')
+      .map(option => {
+        const clonedOption = option.cloneNode(true) as SynOption;
+
+        // Check if the option has a syn-optgroup as parent
+        const hasOptgroup = option.parentElement?.tagName.toLowerCase() === 'syn-optgroup';
+        if (!hasOptgroup) {
+          return clonedOption;
+        }
+
+        const optgroup = option.parentElement as SynOptGroup;
+        const filteredOptgroup = optgroups.find((el) => el.id === optgroup.id);
+
+        // Check if the optgroup was already added to the filteredOptions.
+        // It should only be added once!
+        if (filteredOptgroup) {
+          filteredOptgroup?.appendChild(clonedOption);
+          return undefined;
+        }
+
+        const clonedOptgroup = optgroup.cloneNode() as SynOptGroup;
+        clonedOptgroup.appendChild(clonedOption);
+        optgroups.push(clonedOptgroup);
+        return clonedOptgroup;
+      })
+      // we need to remove the undefined values here
+      .filter(el => el) as Array<SynOption | SynOptGroup>;
```

Comment:
Theres a nice shortcut here that I often use and may be seen as more "elegant" by some:


```suggestion
      .filter(Boolean) as Array<SynOption | SynOptGroup>;
```